powerdns/tests/test_ownership.py

```diff
@@ -0,0 +1,88 @@
+"""Tests for record/domain ownership"""
+
+import base64
+
+from django.contrib.auth.models import User
+from django.core import mail
+from django.test import TestCase
+from rest_framework.test import APIClient
+
+from powerdns.tests.utils import DomainFactory
+
+
+class TestOwnershipBase(TestCase):
+    """Base test class creating some users."""
+
+    def setUp(self):
+        self.user1 = User.objects.create_superuser(
+            'user1', 'user1@example.com', 'password'
+        )
+        self.user2 = User.objects.create_superuser(
+            'user2', 'user2@example.com', 'password'
+        )
+        userpass = base64.b64encode(
+            '{}:{}'.format(self.user1.username, 'password').encode(
+                'ascii'
+            )
+        ).decode('ascii')
+        self.client = APIClient(HTTP_AUTHORIZATION='Basic {}'.format(userpass))
+        mail.outbox = []
+
+
+class TestDomainOwnership(TestOwnershipBase):
+    """Tests for domain ownership"""
+
+    def test_auto_user(self):
+        """Domain owner is set to current user if no owner is specified"""
+        request = self.client.post(
+            '/api/domains/',
+            data={'name': 'owned.example.com'},
+        )
+        self.assertEqual(request.data['owner'], 'user1')
+        self.assertEquals(len(mail.outbox), 0)
+
+    def test_explicit_user(self):
+        """Domain owner is set to explicitly set value"""
+        request = self.client.post(
+            '/api/domains/',
+            data={'name': 'owned.example.com', 'owner': 'user2'},
+        )
+        self.assertEqual(request.data['owner'], 'user2')
+        self.assertEquals(len(mail.outbox), 1)
+
+
+class TestRecordOwnership(TestOwnershipBase):
+    """Tests for record ownership"""
+
+    def setUp(self):
+        super(TestRecordOwnership, self).setUp()
+        self.domain = DomainFactory(name='owned.example.com')
+
+    def test_auto_user(self):
+        """Record owner is set to current user if no owner is specified"""
+        request = self.client.post(
+            '/api/records/',
+            data={
+                'domain': '/api/domains/{}/'.format(self.domain.pk),
+                'type': 'CNAME',
+                'name': 'www.owned.example.com',
+                'content': 'blog.owned.example.com',
+            },
+        )
+        self.assertEqual(request.data['owner'], 'user1')
+        self.assertEquals(len(mail.outbox), 0)
+
+    def test_explicit_user(self):
+        """Record owner is set to explicitly set value"""
+        request = self.client.post(
+            '/api/records/',
+            data={
+                'domain': '/api/domains/{}/'.format(self.domain.pk),
+                'type': 'CNAME',
+                'name': 'www.owned.example.com',
+                'content': 'blog.owned.example.com',
+                'owner': 'user2',
+            },
+        )
+        self.assertEqual(request.data['owner'], 'user2')
```

Comment:
Your assertions are very similar in the whole file - maybe some custom assertion?

E.g.:
``_assert(self, expected_user, expected_mailbox_length)``